app/models/wiki.rb

```
@@ -1,20 +1,18 @@
 class Wiki < ActiveRecord::Base
+  #has_many :collaborators
+  has_many :collaborators, dependent: :destroy
+  has_many :users, through: :collaborators
   belongs_to :user
 
-  validates :title, length: { minimum: 5 }, presence: true
-  validates :body, length: { minimum: 20 }, presence: true
-
-  validates :user, presence: true
-
   # check to see if a user is present of signed in.
   # If it is a premium or admin user, show all. Else, show all wikis that are public
-  scope :visible_to, -> (user){ user && ((user.premium?) || (user.admin?)) ? where(private: false) + where(private: true, user: user) : where(private: false) }
+  #scope :visible_to, -> (user){ user && ((user.premium?) || (user.admin?)) ? where(private: false) + where(private: true, user: user) : where(private: false) }
 
   # make private wikis public
-  def self.make_public(user)
-    user.wikis.each do |w|
-      w.update_attribute :private, false
-    end
+  def make_public
+    self.private = false
+    save
```

I would make this a save! so that it would throw an exception if it failed and would not fail silently.

---

Or use #update_attribute so I wouldn't have two lines of code here. The `self` keyword is not necessary here, too.